tactics/induction.ml

```diff
@@ -1031,7 +1031,7 @@ let find_induction_type env sigma isrec elim hyp0 sort = match elim with
   let is_elim, ind, typ = guess_elim_shape env sigma isrec sort hyp0 in
   (* We drop the scheme and elimc/elimt waiting to know if it is dependent, this
     needs no update to sigma at this point. *)
-  let elim = if is_elim then ElimOver (isrec, hyp0, ind) else CaseOver (hyp0, ind) in
+  let elim = if is_elim then ElimOver (hyp0, ind) else CaseOver (hyp0, ind) in
```

Comment:
I don't see how isrec is statically known to be true here

Reply:
Then how about building the `elim` there instead of returning `is_elim`

Reply:
I want to keep the code separate, because eventually CaseOver will be a fully different tactic.